app/models/container_label_tag_mapping.rb

```diff
@@ -9,91 +9,81 @@ class ContainerLabelTagMapping < ApplicationRecord
   # - When `label_value` is NULL, we map this name with any value to per-value tags.
   #   In this case, `tag` specifies the category under which to create
   #   the value-specific tag (and classification) on demand.
-  #   We then also add a specific `label_value`->specific `tag` mapping here.
+  #
+  # All involved tags must also have a Classification.
 
   belongs_to :tag
 
-  def self.drop_cache
-    @hash_all_by_name_type_value = nil
-  end
-
-  # Returns {[name, type, value] => [tag, ...]}}} hash.
-  def self.hash_all_by_name_type_value
-    unless @hash_all_by_name_type_value
-      @hash_all_by_name_type_value = {}
-      includes(:tag).find_each { |m| load_mapping_into_hash(m) }
+  # Returns {[name, type, value] => [tag_id, ...]}}} hash.  Pass it to map_* methods.
+  def self.cache
```

Comment:
@cben isn't it easier if the "cache" (which I would stop referring to as cache in the code) was an object of another class? That object could also contain the `map_label`, etc.

We can refactor it later but if you feel that this change would simplify things you can get it now.